Sources/StreamChatUI/Utils/String+Extensions.swift

```diff
@@ -8,3 +8,42 @@ extension StringProtocol {
     var firstUppercased: String { prefix(1).uppercased() + dropFirst() }
     var firstLowercased: String { prefix(1).lowercased() + dropFirst() }
 }
+
+// MARK: - Emoji
+
+extension String {
+    /// Whether the string is only a single emoji
+    var isSingleEmoji: Bool {
+        count == 1 && containsEmoji
+    }
+    
+    /// Whether the string contains any emoji
+    var containsEmoji: Bool {
+        unicodeScalars.contains { $0.isEmoji }
+    }
+}
+
+// MARK: - UnicodeScalar
+
+private extension UnicodeScalar {
+    var isEmoji: Bool {
+        switch value {
+        case 0x1f600...0x1f64f, // Emoticons
+             0x1f300...0x1f5ff, // Misc Symbols and Pictographs
+             0x1f680...0x1f6ff, // Transport and Map
+             0x1f1e6...0x1f1ff, // Regional country flags
+             0x2600...0x26ff, // Misc symbols
+             0x2700...0x27bf, // Dingbats
+             0xe0020...0xe007f, // Tags
+             0xfe00...0xfe0f, // Variation Selectors
+             0x1f900...0x1f9ff, // Supplemental Symbols and Pictographs
+             127_000...127_600, // Various asian characters
+             65024...65039, // Variation selector
+             9100...9300, // Misc items
+             8400...8447: // Combining Diacritical Marks for Symbols
+            return true
```

Comment:
Where's does this come from? :)

Reply:
Updated the source code to include some enhancements that were introduced in Swift 5. Also updated the relevant references